test/test_integration_hostfactory.py

```diff
@@ -44,8 +54,141 @@ def test_hostfactory_without_id_returns_menu(self):
+                                  '--duration-days', '1', '--duration-hours', '1', '--duration-minutes', '1'])
+
+        self.assertIn('[\n    {\n        "cidr": [],\n'
+                      f'        "expiration": "{(datetime.utcnow().replace(microsecond=0) + timedelta(days=1, hours=1, minutes=1)).isoformat()}Z",\n'
+                      '        "token":', output)
+
+    def test_hostfactory_with_zero_value_duration_will_raise_error(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory',
+                                  '--duration-days', '0', '--duration-hours', '0', '--duration-minutes', '0'])
+
+        self.assertIn("Failed to execute command. Reason: Parameter", output)
+
+    def test_hostfactory_with_only_days_duration_flags_returns_correct_response(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory',
+                                  '--duration-days', '365'])
+
+        self.assertIn('[\n    {\n        "cidr": [],\n'
+                      f'        "expiration": "{(datetime.utcnow().replace(microsecond=0) + timedelta(days=365)).isoformat()}Z",\n'
+                      '        "token":', output)
+
+    def test_hostfactory_with_only_hours_duration_flags_returns_correct_response(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory',
+                                  '--duration-hours', '24'])
+
+        self.assertIn('[\n    {\n        "cidr": [],\n'
+                      f'        "expiration": "{(datetime.utcnow().replace(microsecond=0) + timedelta(hours=24)).isoformat()}Z",\n'
+                      '        "token":', output)
+
+    def test_hostfactory_with_only_minutes_duration_flags_returns_correct_response(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory',
+                                  '--duration-minutes', '60'])
+
+        self.assertIn('[\n    {\n        "cidr": [],\n'
+                      f'        "expiration": "{(datetime.utcnow().replace(microsecond=0) + timedelta(minutes=60)).isoformat()}Z",\n'
+                      '        "token":', output)
+
+    def test_hostfactory_with_negative_duration_days_flags_raises_error(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory',
+                                  '--duration-days', '-1'], exit_code=1)
+
+        self.assertIn("Failed to execute command. Reason: Parameter", output)
+
+    def test_hostfactory_with_count_returns_correct_response(self):
```

Comment:
Change the test to verify 10 instead of 3

Reply:
this isn't final so will add the logic and test when it is